### PR TITLE
Update self-hosted docs to include service metadata endpoint

### PR DESCRIPTION
**Pull request by komalali**
What it says on the tin. Also reformatted markdown tables.

Fixes: https://github.com/pulumi/pulumi-service/issues/11649

### CODE REVIEW

File: themes/default/content/docs/guides/self-hosted/components/api.md [hunk 1]
```diff
@@ -91,21 +91,23 @@ If your Minio configuration doesn't require a region, you should still specify t
...
+| PULUMI_CHECKPOINT_BLOB_STORAGE_ENDPOINT       | The storage endpoint for persisting stack state/checkpoint. The value takes the format: `azblob://<blob-container>`.                                                                                                    |
+| PULUMI_POLICY_PACK_BLOB_STORAGE_ENDPOINT      | The storage endpoint for persisting published policy packs. The value takes the format: `azblob://<blob-container>`.                                                                                                    |
+| PULUMI_ENGINE_EVENTS_BLOB_STORAGE_ENDPOINT    | The storage endpoint for persisting large engine events. We recommend configuring this value to improve performance and reduce the size of the MySQL database. The value takes the format: `azblob://<blob-container>`. |
+| PULUMI_SERVICE_METADATA_BLOB_STORAGE_ENDPOINT | The storage endpoint for persisting large config values. We recommend configuring this value if your programs use configuration values exceeding 64kb. The value takes the format: `azblob://<bucket-name.`.            |
```
Comment (komalali) [inline themes/default/content/docs/guides/self-hosted/components/api.md:108]: This is the relevant change

File: themes/default/content/docs/guides/self-hosted/components/api.md [hunk 2]
```diff
@@ -91,21 +91,23 @@ If your Minio configuration doesn't require a region, you should still specify t
...
+| PULUMI_POLICY_PACK_BLOB_STORAGE_ENDPOINT      | The storage endpoint for persisting published policy packs. The value takes the format: `s3://<bucket-name>`. Similar to `PULUMI_CHECKPOINT_BLOB_STORAGE_ENDPOINT`, this also supports query-params.                                                                                                    |
+| PULUMI_ENGINE_EVENTS_BLOB_STORAGE_ENDPOINT    | The storage endpoint for persisting large engine events. We recommend configuring this value to improve performance and reduce the size of the MySQL database. The value takes the format: `s3://<bucket-name>`. Similar to `PULUMI_CHECKPOINT_BLOB_STORAGE_ENDPOINT`, this also supports query-params. |
+| PULUMI_SERVICE_METADATA_BLOB_STORAGE_ENDPOINT | The storage endpoint for persisting large config values. We recommend configuring this value if your programs use configuration values exceeding 64kb. The value takes the format: `s3://<bucket-name.`. Similar to `PULUMI_CHECKPOINT_BLOB_STORAGE_ENDPOINT`, this also supports query-params.         |
```
Comment (komalali) [inline themes/default/content/docs/guides/self-hosted/components/api.md:99]: This is the relevant change

File: themes/default/content/docs/guides/self-hosted/components/api.md [hunk 3]
```diff
@@ -49,25 +49,25 @@ This container runs an HTTP server which provides the APIs needed by the Console
...
 It's possible to split apart the HTTP Server from the Background Jobs functionality to distribute reponsibility across multiple instances. The function mode determines the mode in which the server runs based upon the env variable `PULUMI_API_CONTAINER_MODE`. If this value is not defined, or doesn't match a valid option, then it defaults to `FULL`.
 
-| Variable Name | Description                  |
-| ------------- | ---------------------------- |
-|	FULL          | Pulumi API & Background Jobs (Default) |
-| API           | Only Pulumi API              |
-| JOBS          | Only Background Jobs         |
+| Variable Name | Description                            |
+|---------------|----------------------------------------|
+| 	FULL         | Pulumi API & Background Jobs (Default) |
```
Comment (blampe) [inline themes/default/content/docs/guides/self-hosted/components/api.md:54]: This guy's going off on a walk...

**Review by blampe — APPROVED** [review]
> Also reformatted markdown tables.

🙏

File: themes/default/content/docs/guides/self-hosted/components/api.md (hunk 3 above)
Comment (komalali) [reply themes/default/content/docs/guides/self-hosted/components/api.md]: Weird! Fixed.

**Pull request merged**